### PR TITLE
UserBase login with userbaseDSN

### PR DESCRIPTION
**Pull request by prajapati-kaushik**

### CODE REVIEW

File: src/Framework/BaseWebApplication.php [hunk 1]
```diff
@@ -786,6 +787,41 @@ protected function getUserProvider()
...
+
+                case 'UserBaseDsn':
+                    if (!$providerConfig['dsn']) {
+                        throw new RuntimeException('Userbase dsn not configured');
+                    }
+                    $dsn = $providerConfig['dsn'];
+                    if (!filter_var($dsn, FILTER_VALIDATE_URL)) {
+                        throw new RuntimeException('DSN is an invalid URL: '.$dsn);
+                    }
+                    $part = parse_url($dsn);
```
Comment (joostfaassen) [inline src/Framework/BaseWebApplication.php]: This code is not necessary, the userbase client library already automatically supports loading username/password from url. See: https://github.com/linkorb/userbase-client-php/blob/master/src/Client.php#L33

I prefer just keeping one Userbase provider.

You can simply change the original 'UserBase' switch case, to check 'dsn' first, only if it does not exist, get url, username and password.

File: src/Framework/BaseWebApplication.php [hunk 2]
```diff
@@ -89,13 +89,13 @@ public function __construct(array $values = array())
                 $token = $app['security.token_storage']->getToken();
                 if ($token) {
                     $user = $token->getUser();
-                    if ($user=='anon.') {
+                    if ('anon.' == $user) {
```
Comment (joostfaassen) [inline src/Framework/BaseWebApplication.php:92]: Can you try to put code-style and functional changes in separate commits please? It's much better for git history, and reviewing.

**Review by joostfaassen — COMMENTED** [review]
@kaushikindianic can you review these comments?

File: src/Framework/BaseWebApplication.php (hunk 1 above)
Comment (prajapati-kaushik) [reply src/Framework/BaseWebApplication.php]: okay

File: src/Framework/BaseWebApplication.php (hunk 2 above)
Comment (prajapati-kaushik) [reply src/Framework/BaseWebApplication.php:92]: That done by the editor when saving file.

**Pull request merged**